### PR TITLE
Add SGD algorithm

### PR DESCRIPTION
**Pull request by jilljenn**

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Why sample.py ?

File: mangaki/mangaki/algo/sgd.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from scipy.sparse import coo_matrix
+from sklearn.metrics import mean_squared_error
+from sklearn.metrics.pairwise import cosine_similarity
+import random
+
+from mangaki.algo.recommendation_algorithm import RecommendationAlgorithm, register_algorithm
+
+
+@register_algorithm('sgd')
+class MangakiSGD(RecommendationAlgorithm):
```
Comment (RaitoBezarius) [inline mangaki/mangaki/algo/sgd.py]: I'd prefer we avoid putting `nb_users` and `nb_works` in the constructor, it is not matching the current logic of all recommendation algorithm, until we move away from `set_params`, we should keep it as-is.

File: mangaki/mangaki/algo/sgd.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self.U = np.random.random((self.nb_users, self.nb_components))
+        self.V = np.random.random((self.nb_works, self.nb_components))
+
+    def fit(self, X, y):
+        for epoch in range(self.nb_iterations):
+            step = 0
+            for (i, j), rating in zip(X, y):
+                if step % 100000 == 0:  # Pour afficher l'erreur de train
+                    y_pred = self.predict(X)
+                    print('Train RMSE (epoch={}, step={}): {}'.format(
```
Comment (RaitoBezarius) [inline mangaki/mangaki/algo/sgd.py]: logger please :)

File: mangaki/mangaki/algo/sgd.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    def predict(self, X):
+        y = []
+        for user_id, work_id in X:
+            y.append(self.predict_one(user_id, work_id))
+        return np.array(y)
+
+    @property
+    def is_serializable(self):
+        return True
```
Comment (RaitoBezarius) [inline mangaki/mangaki/algo/sgd.py]: The bias won't be serialized, right? Are you sure you still want to serialize it?

File: mangaki/mangaki/algo/sgd.py (hunk 1 above)
Comment (jilljenn) [reply mangaki/mangaki/algo/sgd.py]: Sorry, it was some leftover of an experiment. Removed.

Comment (jilljenn) [reply mangaki/mangaki/algo/sgd.py]: Also the reason was: it was better to initialize the parameters in `__init__`, but for this we need the `nb_users` and `nb_works` before it is specified using `set_params`.

File: mangaki/mangaki/algo/sgd.py (hunk 2 above)
Comment (jilljenn) [reply mangaki/mangaki/algo/sgd.py]: :'( https://www.reddit.com/r/Python/comments/387oog/when_will_logging_use_newstyle_string_formatting/

File: mangaki/mangaki/algo/sgd.py (hunk 3 above)
Comment (jilljenn) [reply mangaki/mangaki/algo/sgd.py]: Actually yes I want! But it will be in another PR, so I put `False`.

**Pull request merged**